packages/e2e-tests/utils/account.js

```diff
@@ -1,99 +1,77 @@
 const {
-    connect,
-    keyStores: { InMemoryKeyStore },
     utils: {
-        KeyPair,
         format: { parseNearAmount },
     },
 } = require("near-api-js");
-const { parseSeedPhrase } = require("near-seed-phrase");
+const BN = require("bn.js");
 
-const { walletNetwork } = require("./config");
+const nearApiJsConnection = require("./connectionSingleton");
+const { getKeyPairFromSeedPhrase } = require("./helpers");
 
-const getDefaultConfig = () => ({
-    networkId: walletNetwork,
-    nodeUrl: process.env.NODE_URL || "https://rpc.testnet.near.org",
-    walletUrl: process.env.WALLET_URL || "https://wallet.testnet.near.org",
-    keyStore: new InMemoryKeyStore(),
-});
-
-function getKeyPairFromSeedPhrase(seedPhrase) {
-    return KeyPair.fromString(parseSeedPhrase(seedPhrase).secretKey);
+class E2eTestAccount {
+    constructor(accountId, seedPhrase, parentNearApiJsAccount) {
+        this.accountId = accountId;
+        this.seedPhrase = seedPhrase;
+        this.parentNearApiJsAccount = parentNearApiJsAccount;
+    }
+    async initialize() {
+        await nearApiJsConnection.setKeyPairFromSeedPhrase(this);
+        await this.connectToNearApiJs();
+        return this;
+    }
+    async connectToNearApiJs() {
+        const near = await nearApiJsConnection.getConnection();
+        this.nearApiJsAccount = await near.account(this.accountId);
+    }
+    async create({ amount } = { amount: "1.0" }) {
+        await this.parentNearApiJsAccount.createAccount(
+            this.accountId,
+            getKeyPairFromSeedPhrase(this.seedPhrase).publicKey,
+            parseNearAmount(amount)
+        );
+        this.isCreated = true;
+        await this.initialize();
+        return this;
+    }
+    spawnRandomSubAccountInstance() {
+        if (!this.nearApiJsAccount) {
+            throw new Error("Account needs to be initialized to spawn sub accounts");
+        }
+        const randomSubaccountId = generateTestAccountId();
+        const randomSubaccountSeedphrase = `${randomSubaccountId} ${process.env.TEST_ACCOUNT_SEED_PHRASE}`;
```

Comment:
Curious about this -- why do prepend the seed phrase with `randomSubaccountId`?  If this is important, we should comment here on why.  It seems a bit out of place given that actual users always enter a 12 word seed phrase conforming to bip39.  If we want unique seed phrases for each account, we can just generate them using `near-seed-phrase` (https://github.com/near/near-wallet/blob/3c8ac4a32750597d4d467d0aa53e4f381e975dc5/packages/frontend/src/components/accounts/SetupSeedPhrase.js#L64)